src/main/java/fruit/shop/service/TransactionStrategy.java

```diff
@@ -0,0 +1,7 @@
+package fruit.shop.service;
+
+import fruit.shop.model.FruitTransaction;
+
+public interface TransactionStrategy {
+    void executeTransactionHandler(FruitTransaction transaction);
```

Comment:
`handleTransaction()` - `executeTransactionHandler` is misleading as you're not passing any handler. Handler is not known to outside world, hence the name says nothing to user.